### DISCUSSION TITLE
Hey, HI! My novel front end consumes your api. Can we be friends?

### DISCUSSION THREAD

**Discussion by aseichter2007** [LostRuins/koboldcpp#541] [General]
I made a really cool tool and I'm super grateful for your work, It's the only model host I have successfully launched on my windows box. I haven't quite groked docker yet, so you're my hero.

I made[ Clipboard Conqueror ](https://github.com/aseichter2007/ClipboardConqueror)
It interfaces with the windows clipboard(and maybe mac, who knows) 
Clipboard Conqueror is an everywhere copilot solution and I invite you to extract value from it as it runs pretty seamlessly alongside your own UI.  I hope that our applications continue to compliment each other in the future, and if you have and feedback or feature requests I will get right on it. 

|||frank| "Frank, help me out here. We have to welcome our new friend." *I say, motioning at him to come closer.* "Am I forgetting anything? This man is a hero!"

copy all above^
Paste:
"Ah, yes, of course. *Frank's eyes light up with realization* I'm happy to meet you, too! It's an honor to have a hero among us. So, let's welcome our new friend and introduce him to the wonderful world of Clipboard Conqueror." *Frank motions for the newcomer to approach*

"As my partner here says, Clipboard Conqueror is a fantastic tool that interfaces with the Windows clipboard, and possibly even Mac, I'm not too sure on that one. But let's focus on what we do know! It's an everywhere copilot solution that's been running seamlessly alongside my UI, and I believe it can do the same for you." *Frank smiles warmly*

"If you ever have any questions or need assistance, feel free to reach out to us. We're always here to help. And, of course, if you have any suggestions or feature requests, we're more than happy to listen and implement them. We're all about making Clipboard Conqueror the best it can be, and your feedback is crucial to that process." *Frank raises his eyebrows, waiting for any questions or comments*

**Comment by LostRuins** [comment]
Interesting idea. Might be a little clunky to use though, I'd imagine shell integration might be more intuitive (e.g. trigger with right click context menu).

**Reply by aseichter2007** [reply]
Try it out mate. It's pretty legit. It runs great alongside your UI, but can cause the context to be recalculated if that is important to you. 
Text boxes become nice context spaces to work in, just leave the |||code| tag setting the agent you want to work against and then ctrl+a then ctrl+c, right arrow, enter, ctrl+v. 
and you can add to the bottom like  

User: {request} and hermes is plenty smart to chat in context pretty nice. 

I agree a menu option might be a good idea, but for personal use the mouse is just in the way. It makes the demo more clear though shift + up arrow and c is just fast and my hands are already on the typer.  I think the demo video is the first time I consciously used the mouse on purpose to copy.   Now A keyboard hotkey I have been considering. Listen for ctrl shift c or something and run with no invoke. Oh jeez that collides with chrome. 

Since ||| is a pretty unlikely token and the invoke can be changed in the settings, I've never accidentally set off the interface, and transporting instructions is easy by overloading the parse engine with too many ||| ||| |||(max 2 or input is not parsed or sent out. )


Edit: plus adding menu context buttons that only do any good with two extra programs running just sounds dirty. even if I made my app a windows service it's still a useless button without a backend running or maybe I can add and remove them at runtime, but that sounds strange and waiting to cause problems. 

Thank you for the feedback, but in this instance, it would be poor I think. Plus the temptation to just add a second button that is a renamed copy that appends an invoke is too tempting, since I am definitely not removing how it invokes right now. It's quite smooth.   If I ever get this to an all in one I will have to do it as a service that loads models at first ask, and then it will be appropriate to start integrating with the shell.
